converters/fromZigbee.js

```diff
@@ -2682,7 +2682,115 @@ const converters = {
             return result;
         },
     },
+    ZNMS12LM_closuresDoorLock_change: {
+        cid: 'closuresDoorLock',
+        type: 'attReport',
+        convert: (model, msg, publish, options) => {
+            const result = {};
+            const lockStatusLookup = {
+                1: 'finger_not_match',
+                2: 'password_not_match',
+                3: 'reverse_lock', // disable open from outside
+                4: 'reverse_lock_cancel', // enable open from outside
+                5: 'locked',
+                6: 'lock_opened',
+                7: 'finger_add',
+                8: 'finger_delete',
+                9: 'password_add',
+                10: 'password_delete',
+                11: 'lock_opened_inside', // Open form inside reverse lock enbable
+                12: 'lock_opened_outside', // Open form outside reverse lock disable
+                13: 'ring_bell',
+                14: 'change_language_to',
+                15: 'finger_open',
+                16: 'password_open',
+            };
+            result.user = null;
+            result.repeat = null;
+            if (msg.data.data['65526']) { // lock final status
+                // Convert data back to hex to decode
+                const data = Buffer.from(msg.data.data['65526'], 'ascii').toString('hex');
+                const command = data.substr(6, 4);
+                if (command === '0301') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'UNLOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0311') {
+                    result.event = lockStatusLookup[4];
+                    result.state = 'LOCK';
+                    result.reverse = 'UNLOCK';
+                } else if (command === '0205') {
+                    result.event = lockStatusLookup[3];
```

Comment:
Could you rename `event` to `action`? In this way we can use the home assistant action sensor which is already in place.

Reply:
Yes, i have already changed it